docs/conf.py

```diff
@@ -119,3 +119,163 @@
 # Tell sphinx-autodoc-typehints to generate stub parameter annotations including
 # types, even if the parameters aren't explicitly documented.
 always_document_param_types = True
+
+
+# -- Patch sphinx.ext.autosummary.generate ----------------------------------
+
+# This patch is needed to make autosummary provide the "annotations"
```

Comment:
Should we split the patch into a separate module for better readability?
That way the `config.py` would remain short and be about config, and the (temporary) patch would live in a separate file.

Reply:
Thanks for bringing this up! That was my original idea but I was hesitant to introduce a new file here, since we agree on this I will be adding this code to a new `sphinx_patch.py` file.

Reply:
Created `conf_sphinx_patch.py`.